pkg/cri/resource-manager/cache/cache.go

```diff
@@ -1110,6 +1128,47 @@ func (cch *cache) setEntry(key string, ptr, obj interface{}) error {
 	return nil
 }
 
+// checkFilePermissions checks permission bits of an existing file.
+func (cch *cache) checkFilePermissions(path string, perm, rejected os.FileMode, what string) error {
+	if info, err := os.Stat(path); err == nil && !info.IsDir() {
+		expected := perm & os.ModePerm // deliberately ignore sticky bit in comparison
+		existing := info.Mode().Perm() & os.ModePerm
+		if rejected != 0 && (existing&rejected) != 0 {
```

Comment:
nit: the first check is not needed, right(?) If rejected==0 then existing&rejected==0

Reply:
Split out permissions check, added missing ownership tests, and removed the extra zero-test.